src/main/java/me/thegiggitybyte/wumpuslink/Commands.java

```diff
@@ -0,0 +1,126 @@
+                setValueString = valueString;
+
+            } else if (valuePrimitive.isBoolean()) {
+                if (valueString.equals("true") | valueString.equals("false")) {
+                    var valueBoolean = Boolean.parseBoolean(valueString);
+                    JsonConfiguration.getUserInstance().setValue(keyString, valueBoolean);
+                    setValueString = String.valueOf(valueBoolean);
+                } else {
+                    messageText = Text.literal("Configuration key '").append(keyString).append("' expects a boolean (true or false)");
+                }
+
+            } else if (valuePrimitive.isNumber()) {
+                try {
+                    var valueInt = Integer.parseInt(valueString);
+                    JsonConfiguration.getUserInstance().setValue(keyString, valueInt);
+                    setValueString = String.valueOf(valueInt);
+                } catch (NumberFormatException intException) {
+                    try {
+                        var valueDouble = Double.parseDouble(valueString);
+                        JsonConfiguration.getUserInstance().setValue(keyString, valueDouble);
+                        setValueString = String.valueOf(valueDouble);
+                    } catch (NumberFormatException doubleException) {
+                        messageText = Text.literal("Configuration key '").append(keyString).append("' expects an integer or float number");
+                    }
+                }
+            } else {
+                throw new SimpleCommandExceptionType(Text.literal("Unhandled value type")).create();
+            }
+        } else {
+            messageText = Text.empty()
+                    .append(Text.literal("✖ ").formatted(Formatting.RED))
+                    .append(Text.literal(keyString).formatted(Formatting.DARK_RED))
+                    .append(Text.literal(" is an invalid configuration key.").formatted(Formatting.RED));
+        }
+
+        if (messageText == null) {
+            JsonConfiguration.getUserInstance().writePendingChanges();
+
+            messageText = Text.empty()
+                    .append(Text.literal("✔ ").formatted(Formatting.GREEN))
+                    .append(Text.literal(keyString + " set to ").formatted(Formatting.WHITE))
+                    .append(Text.literal(setValueString).formatted(Formatting.YELLOW));
+        }
+
+        ctx.getSource().sendFeedback(messageText, false);
+        return Command.SINGLE_SUCCESS;
+    }
+
+    private static CompletableFuture<Suggestions> getConfigKeySuggestions(CommandContext<ServerCommandSource> ctx, SuggestionsBuilder builder) {
+        for (var key : JsonConfiguration.getDefaultInstance().getKeys())
+            builder.suggest(key, Text.literal("key"));
+
+        return builder.buildFuture();
+    }
+
+    private static CompletableFuture<Suggestions> getConfigValueSuggestions(CommandContext<ServerCommandSource> ctx, SuggestionsBuilder builder) {
```

Comment:
My main concern with adding the config command from SleepWarp is that completion suggestions will expose the Discord token to anybody with access to the command. Perhaps there should be a check for that here which will return nothing?

Reply:
Also, now that I'm thinking about it, setting the Discord token though the command would also not restart the Discord bot. I think there still needs to be some sort of proper config reload mechanism.

Reply:
> Also, now that I'm thinking about it, setting the Discord token though the command would also not restart the Discord bot. I think there still needs to be some sort of proper config reload mechanism.

Can't we just reconnect bot, and it would use updated values?